install/migrations/update_10.0.7_to_10.0.8/cartridgelevelalert.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * @var DB $DB
+ * @var Migration $migration
+ */
+
+$migration->addField('glpi_cartridgeitems', 'type_tag', "varchar(255)");
+$migration->addField('glpi_cartridgeitems', 'warn_level', "int unsigned NOT NULL DEFAULT '0'");
+$migration->addKey('glpi_cartridgeitems', 'type_tag', 'type_tag');
+
+$migration->addField('glpi_entities', 'printer_cartridge_levels_alert_repeat', "int NOT NULL DEFAULT '-2'");
+
+// Add crontask for low level notifications
+$crontask = new CronTask();
+if (empty($crontask->find(['name' => 'PrinterCartridgeLevelAlert']))) {
+    $cron_added = CronTask::register(
+        'PrinterCartridgeLevelAlert',
+        'PrinterCartridgeLevelAlert',
+        1 * DAY_TIMESTAMP,
+        [
+            'state' => CronTask::STATE_DISABLE,
+            'mode' => CronTask::MODE_EXTERNAL
+        ]
+    );
+    if (!$cron_added) {
+        die("Can't add PrinterCartridgeLevelAlert cron");
+    }
+}
+
+// Insert notification
```

Comment:
Notifications must also be added in the case of a fresh install too (see ```install/empty_data.php```)

Reply:
Done.